shared/src/ledger/storage/merkle_tree.rs

```diff
@@ -273,18 +272,7 @@ impl<H: StorageHasher + Default> MerkleTree<H> {
         // Get a proof of the sub tree
         let hashed_sub_key = H::hash(&sub_key.to_string());
         let cp = subtree.non_membership_proof(&hashed_sub_key)?;
-        // Replace the key with the non-hashed key for the verification
-        let sub_proof = match cp.proof.expect("The proof should exist") {
-            Ics23Proof::Nonexist(nep) => CommitmentProof {
-                proof: Some(Ics23Proof::Nonexist(NonExistenceProof {
-                    key: sub_key.to_string().as_bytes().to_vec(),
-                    ..nep
-                })),
-            },
-            // the proof should have a NonExistenceProof
-            _ => unreachable!(),
-        };
-        self.get_proof(key, sub_proof)
+        self.get_proof(key, cp)
```

Comment:
Just a comment.
This fix is a workaround for the added test. Non-existence proof verification will still fail in IBC VP because the proof spec is the same as existence proof verification.
The issue has been opened: #1007